lib/rules/function-url-quotes/index.mjs

```diff
@@ -54,48 +58,83 @@ const rule = (primary, secondaryOptions, context) => {
 		root.walkDecls(checkDeclParams);
 
 		/**
-		 * @param {import('postcss').Declaration} decl
+		 * @param {Declaration|AtRule} node
+		 * @param {FunctionNode} fn
+		 * @param {'add'|'remove'} mutateType
+		 * @param {ParsedValue} parsedValue
+		 */
+		function fixer(node, fn, mutateType, parsedValue) {
+			return () => {
+				switch (mutateType) {
+					case 'add':
+						addQuotes(fn);
+						break;
+					case 'remove':
+						removeQuotes(fn);
+						break;
+				}
+
+				switch (node.type) {
+					case 'decl':
+						node.value = parsedValue.toString();
+						break;
+					case 'atrule':
+						node.params = parsedValue.toString();
+						break;
+				}
+			};
+		}
+
+		/** @typedef {[Declaration|AtRule, FunctionNode, 'add'|'remove']} ArgsArray */
+		/** @typedef {Array<[Omit<Problem, 'node'>, ArgsArray]>} ReportQueue */
+
+		/**
+		 * @param {AtRule|Declaration} node
+		 * @param {string} source
+		 * @param {number} startIndex
 		 */
+		function complain(node, source, startIndex) {
+			/** @type {ReportQueue} */
+			const reportQueue = [];
+			const parsedValue = functionArgumentsSearch(source, /^url$/i, (args, index, funcNode) => {
+				const problem = checkArgs(args, startIndex + index, funcNode);
+
+				if (problem) {
+					reportQueue.push([
+						problem,
+						[node, funcNode, problem.message.startsWith('Expected') ? 'add' : 'remove'],
```

Comment:
[suggest] I want to avoid depending on the message (`message.startsWith('Expected')`) for maintainability. Instead, how about adding the message type? For example:

```diff
--- a/lib/rules/function-url-quotes/index.mjs
+++ b/lib/rules/function-url-quotes/index.mjs
@@ -60,7 +60,7 @@ const rule = (primary, secondaryOptions) => {
 		/**
 		 * @param {Declaration|AtRule} node
 		 * @param {FunctionNode} fn
-		 * @param {'add'|'remove'} mutateType
+		 * @param {keyof messages} messageType
 		 * @param {ParsedValue} parsedValue
 		 */
 		function fixer(node, fn, mutateType, parsedValue) {
```